src/utils/GraphQLHints.js

```diff
@@ -0,0 +1,385 @@
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import {
+  isInputType,
+  isCompositeType,
+  isAbstractType,
+  getNullableType,
+  getNamedType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+  GraphQLList,
+  GraphQLBoolean,
+  doTypesOverlap,
+} from 'graphql';
+import {
+  SchemaMetaFieldDef,
+  TypeMetaFieldDef,
+  TypeNameMetaFieldDef,
+} from 'graphql/type/introspection';
+
+import forEachState from './forEachState';
+import hintList from './hintList';
+import objectValues from './objectValues';
+
+import runParser from './runParser';
+import { LexRules, ParseRules, isIgnored } from './Rules';
+
+// Finds all fragment definition ASTs in a source.
+function getFragmentDefinitions(sourceText) {
```

Comment:
this should move lower in the file to where it was before so it's more clear what this file exports